src/components/media-views.js

```diff
@@ -1216,8 +1216,15 @@ AFRAME.registerComponent("media-pdf", {
         this.el.setAttribute("media-pager", { maxIndex: this.pdf.numPages - 1 });
       }
 
-      const page = await this.pdf.getPage(index + 1);
-      if (src !== this.data.src || index !== this.data.index) return;
+      if (!this.pdf) {
+        // The pdf may not have been loaded yet, this can happen with initial
+        // render followed with a networked update with index > 0
+        // (not the first page of the pdf)
+        return;
+      }
+      // Use here this.data.index and not index to have the networked update.
+      const page = await this.pdf.getPage(this.data.index + 1);
+      if (src !== this.data.src) return;
```

Comment:
I removed the "index !== this.data.index" check. We want to render the pdf completely.
The code flows like this:
- first call with `src: test.pdf; index:0`
- pdf is loading (first call)
- second call with `src: test.pdf; index:1` , `(src !== oldData.src) is false` and `this.pdf` is undefined, don't run further
- on the first call, pdf is loaded, `index` equals `0`, `this.data.index` equals `1`, do the rendering with updated `this.data.index`

Reply:
I believe the `index !== this.data.index` check is still needed since `this.pdf.getPage` is async. If you are paging through quickly you could have a race condition where a previous page ends up finishing after the current page, and you do not want to use that result